packages/ui/react-ui-menu/src/testing/index.ts

```diff
@@ -67,10 +66,15 @@ export const createNestedActionGraph = (groupParams?: CreateActionsParams, param
     // @ts-ignore
     graph._addNodes([group as NodeArg<any>, ...(actions as NodeArg<any>[])]);
     // @ts-ignore
-    graph._addEdges(actions.map((action) => ({ source: group.id, target: action.id })));
+    graph._addEdges([
+      { source: 'root', target: group.id },
+      ...actions.map((action) => ({ source: group.id, target: action.id })),
+    ]);
     void graph.expand(group);
   });
-  return { graph, topLevelActions: actionGroups as ToolbarItem[] };
+  const resolveGroupItems = (groupNode?: MenuItemGroup) =>
```

Comment:
I think we should unify this logic with that which is factored out of the editor

Reply:
I’ll see what shakes out and make this a model of how to use that, certainly.